metadata-models/src/main/pegasus/com/linkedin/dataprocess/DataProcessInfo.pdl

```diff
@@ -0,0 +1,19 @@
+namespace com.linkedin.dataprocess
+
+import com.linkedin.common.DatasetUrn
+
+/**
+ * The inputs and outputs of this data process
+ */
+record DataProcessInfo {
+
+  /**
+   * the inputs of the data process
+   */
+  inputs: optional array[DatasetUrn]
```

Comment:
If inputs and outputs in one aspect, then you need to know both before you can add the aspect. It would be better to separate them out into separate aspects like below:

```
record DataProcessInput {

  inputs: optional array[DatasetUrn]
}
```

```
record DataProcessOutput {

  inputs: optional array[DatasetUrn]
}
```

I would keep the `DataProcessInfo` aspect only for basic metadata of a data process.

The advantage of separating input and output is that they can be updated independently (for example at process  or during read you can send MCE for input and at process end/write/sink time, the outputs can be updated independently.

Reply:
OK. Would it be OK that leaves this on the next PR? This changes definitely are related to many changes.

Reply:
It's okay to do it in a next PR.